tests/test_qnode.py

```diff
@@ -226,6 +226,36 @@ def capabilities(cls):
         res = QNode.get_best_method(dev, "another_interface")
         assert res == (qml.gradients.finite_diff, {}, dev)
 
+    def test_best_method_str(self, monkeypatch):
+        """Test that the method for determining the best diff method string
+        for a given device and interface works correctly"""
+        dev = qml.device("default.qubit", wires=1)
+        monkeypatch.setitem(dev._capabilities, "passthru_interface", "some_interface")
+        monkeypatch.setitem(dev._capabilities, "provides_jacobian", True)
+
+        # device is top priority
+        res = QNode.get_best_method_str(dev, "another_interface")
+        assert res == "device"
+
+        # backprop is next priority

```

Comment:
This seems to be a slightly different test case from the previous one. It can be good to separate the cases (mostly) based on `assert` statements. It can help with readability for example when running the test suite, as each failing test case would have a different name.

Reply:
I agree. I'll also split up the original `test_best_method` as well.